doc/nrf/protocols/matter/end_product/security.rst

```diff
@@ -1,16 +1,109 @@
 .. _ug_matter_device_security:
 
-Securing production devices
-###########################
+Security
+########
 
 .. contents::
    :local:
-   :depth: 2
+   :depth: 3
+
+Nordic Matter samples leverage security features supported in the |NCS| that can be divided into three major categories:
+
+* Cryptography
+* Trusted storage
+* Securing production devices
+
+In the following sections you will learn more details about each listed category.
+
+Cryptography
+************
+
+Depending on the networking backend, the |NCS| Matter samples currently use the following APIs to implement cryptographic operations:
+
+* PSA Cryptography API for Thread networking.
+* Mbed TLS for Wi-Fi networking.
+
+Both APIs are integrated in the :ref:`nrf_security` library.
+This library offers various configuration possibilities and backends that can be employed to implement :ref:`cryptographic operations <cryptographic_operations_in_ncs>`.
+You can find an overview of the cryptography layer configuration supported for each |NCS| Matter-enabled platform in the `Matter platforms security support`_ section.
+
+Trusted storage
+***************
+
+:ref:`trusted_storage_in_ncs` is a security mechanism designed to securely store and manage sensitive data, such as cryptographic keys, device credentials, and configuration data.
+Such component is essential for IoT devices and it allows to implement secure communication between devices within the Matter ecosystems.
+
+Depending on the hardware capabilities of the used platform, the Trusted Storage can utilize one of the following backends:
+
+* `ARM TrustZone`_ technology and hardware-accelerated firmware isolation
+*  The :ref:`trusted_storage_readme` |NCS| software library
+
+Currently all :ref:`matter_samples` in the |NCS| use the trusted storage library as a trusted storage backend for all supported platforms.
+
+The trusted storage library provides confidentiality, integrity, and authenticity for the assets it operates on.
+It handles sensitive data in two contexts:
+
+* Volatile - Before data is forwarded to the non-volatile storage and after it is retrieved from the non-volatile storage.
+* Non-volatile - While data is being written to the non-volatile memory.
+
+In the case of the volatile context, the trusted storage library leverages the Authenticated Encryption with Associated Data (AEAD) encryption backend (:kconfig:option:`CONFIG_TRUSTED_STORAGE_BACKEND_AEAD`).
+AEAD can be configured with the set of Kconfig options described in the library's :ref:`trusted_storage_configuration` section.
+
+An important setting, that depends on the hardware platform in use, is the way of generating the AEAD keys.
+The recommended and the most secure option is to use :ref:`lib_hw_unique_key` (HUK) library.
+HUK support is automatically enabled with the :kconfig:option:`CONFIG_TRUSTED_STORAGE_BACKEND_AEAD_KEY_DERIVE_FROM_HUK` Kconfig option for compatible platforms.
+.. note::
```

Comment:
```suggestion
HUK support is automatically enabled with the :kconfig:option:`CONFIG_TRUSTED_STORAGE_BACKEND_AEAD_KEY_DERIVE_FROM_HUK` Kconfig option for compatible platforms.

.. note::
```
Breaks doc build.

Reply:
Yep, I will do rebase anyway. @wiba-nordic  FYI, To facilitate review process I will add two commits, one will incorporate the original contribution + your change requests and the second will modify information regarding HUK. I can then make one final rebase eventually.